include/nuttx/mutex.h

```diff
@@ -214,7 +224,7 @@ static inline bool nxmutex_is_locked(FAR mutex_t *mutex)
 
 static inline int nxmutex_unlock(FAR mutex_t *mutex)
 {
-  return nxsem_post(mutex);
+  return _SEM_POST(mutex);
```

Comment:
```suggestion
  int ret;
  
   ret = _SEM_POST(mutex);
   if (ret < 0)
     {
       return _SEM_ERRNO(ret);
     }
     
   return ret;
```

Reply:
@pkarashchenko nxmutex is designed for kernel only, the userspace should use pthread_mutex_t instead, so we don't need call _SEM_POST/_SEM_WAIT in this file, I think.

Reply:
Do you plan to integrate this suggestion?